### PR TITLE
Remove unstable marker for append-only features

### PR DESCRIPTION
**Pull request by mziolekda**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by tudor-da — APPROVED** [review]

File: ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala [hunk 1]
```diff
@@ -513,40 +513,40 @@ object Config {
           )
 
         // TODO append-only: remove after removing support for the current (mutating) schema
-        opt[Unit]("index-append-only-schema-unsafe")
+        opt[Unit]("index-append-only-schema")
           .optional()
           .hidden()
           .text(
-            s"Use the append-only index database with parallel ingestion. Highly unstable. Should not be used in production."
+            s"Use the append-only index database with parallel ingestion."
```
Comment (miklos-da) [inline ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala:520]:
```suggestion
            "Use the append-only index database with parallel ingestion."
```

File: ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala [hunk 2]
```diff
@@ -513,40 +513,40 @@ object Config {
...
           )
           .action((_, config) => config.copy(enableAppendOnlySchema = true))
 
-        opt[Unit]("mutable-contract-state-cache-unsafe")
+        opt[Unit]("mutable-contract-state-cache")
           .optional()
           .hidden()
           .text(
-            "Experimental contract state cache for command execution. Must be enabled in conjunction with index-append-only-schema-unsafe. Should not be used in production."
+            "Contract state cache for command execution. Must be enabled in conjunction with index-append-only-schema."
```
Comment (miklos-da) [inline ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala:528]: If these must be used in conjunction, why not have a single command-line option e.g. `indexer-enable-parallel-ingestion`?

File: ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala [hunk 3]
```diff
@@ -513,40 +513,40 @@ object Config {
           )
 
         // TODO append-only: remove after removing support for the current (mutating) schema
```
Comment (miklos-da) [inline ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala:515]: Do you still need to keep this TODO around?

Comment (mziolekda) [reply ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala:515]: mutating schema is still around. We will be able to get rid of i when everyone starts using append-only

File: ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala (hunk 2 above)
Comment (mziolekda) [reply ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala:528]: They don't have to be used in conjunction, append-only is fine to be set alone. It is a prerequisite for mutable-state-cache

File: ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala (hunk 1 above)
Comment (mziolekda) [reply ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala:520]: Sorry, the PR went in on automerge before I was able to react

**Pull request merged**